### ISSUE TITLE
Update to 1.5

### ISSUE THREAD

**Issue by bentruyman** [YungSang/boot2docker-vagrant-box#22]
http://blog.docker.com/2015/02/docker-1-5-ipv6-support-read-only-containers-stats-named-dockerfiles-and-more/

**Comment by tonivdv** [comment]
:+1:

**Comment by ailispaw** [comment]
If you don't need VirtualBox FS, I offer only-docker.box instead.
https://atlas.hashicorp.com/ailispaw/boxes/only-docker

**Comment by tonivdv** [comment]
Hey @YungSang ,

You probably got other things on your head, so do you want me to make a PR with the version bump?

Cheers

**Comment by FelikZ** [comment]
:+1:

**Comment by YungSang** [comment]
Sorry for this late reply and telling this.
My activities as YungSang have been shutdown since the end of last year.
I won't update this repo any more.

@tonivdv Thank you for your offering.
Please create your own repo and box for that.

Notes: yungsang/boot2docker may need the custom boot2docker.iso built with https://github.com/YungSang/boot2docker.

https://github.com/wearableintelligence/boot2docker-vagrant-box
This repo may help you to build boot2docker.box with the original boot2docker.iso, though it's for parallels.

**Comment by tonivdv** [comment]
Ah that's sad news :(

Thanks for the effort in the past and good luck in your new adventures! Cheers

**Comment by YungSang** [comment]
@tonivdv Thank you so much. Good luck to you, too.

**Comment by tonivdv** [comment]
Should others be interested, following is maintained and working well https://github.com/dduportal/boot2docker-vagrant-box

**Comment by thedug** [comment]
tonivdv - is there a binary available or did you have to build it yourself?

**Comment by tonivdv** [comment]
@thedug everything is available. You simply need to do 

``` bash
$ vagrant init dduportal/boot2docker
$ vagrant up
```

And if you need nfs, check https://github.com/dduportal/boot2docker-vagrant-box/issues/10

**Comment by thedug** [comment]
When I try to connect I get this error:

FATA[0000] Get http://localhost:2375/v1.17/images/json: EOF. Are you trying to connect to a TLS-enabled daemon without TLS?

$ echo $DOCKER_HOST
tcp://localhost:2375

**Comment by FelikZ** [comment]
@thedug check if guest machine forwarding correctly forwarding 2375 and via `vagrant ssh` check `netstat -nltp` if docker daemon listens to this port.

**Comment by thedug** [comment]
dduportal is not disabling tls like yungsang. I was able to ssh in and manually disable but I don't want to have to make the entire dev team do that.  I'm currently trying to think of a way to do that from the Vagrant file.

Alternatively I setting up the certs from the Vagrant file or with a script would be acceptable as well.

**Comment by tonivdv** [comment]
Hey @thedug ,

You need to add next to the Vagrant file a file called bootlocal.sh with following content:

```
# Regenerate certs for the newly created Iprivate network IP
sudo /etc/init.d/docker restart
# Copy tls certs to the vagrant share to allow host to use it
sudo cp -r /var/lib/boot2docker/tls /vagrant/
```

This file will get called on each vagrant up, and set correct TLS certificats

Then change your export to:

export DOCKER_CERT_PATH=<Absolute path to your vagrant share on host>/tls
export DOCKER_HOST=tcp://127.0.0.1:2376

You could still have permission issue, because by default this box creates shares under the user 'root'. To circumvent that add following in your vagrant file:

``` ruby
config.nfs.map_uid = Process.uid   
config.nfs.map_gid = Process.gid 
```

I have it working fine, hope you'll be able to do it too.

Cheers

**Comment by thedug** [comment]
@tonivdv

This gets exported on the OSX host right? If so will the dir exist in the root? Seems odd.
It will probably exist as a sibbling to the project.

export DOCKER_CERT_PATH=/tls

This is not great as it is not easily scriptable if the path is different for every user.

Disabling TLS seems better and reason not to?

**Comment by tonivdv** [comment]
> Disabling TLS seems better and reason not to?

No reason, having TLS disable feature is probably the way to go.

> This gets exported on the OSX host right? If so will the dir exist in the root? Seems odd.
> It will probably exist as a sibbling to the project.

Not sure if we are on the same line here :). Let me give you a concrete example.

Assume your project containing the `Vagrantfile` and `bootlocal.sh` is situated in `/Users/dummy/project/`. When you do `vagrant up` with the default configuration it will share that directory on the guest in `/vagrant/` and execute the `bootlocal.sh` file which will create in the guest a `tls` directory. So on guest you will have `/vagrant/tls/...` and on host `/Users/dummy/project/tls/...` .

Therefore in your .profile or .bash_profile you will set the export to

``` bash
export DOCKER_CERT_PATH=/Users/dummy/project/tls
export DOCKER_HOST=tcp://127.0.0.1:2376
```

So you can script that during an initial set-up, as it's always related to current directory where the Vagrantfile is situated. But you are free to change the bootlocal.sh file to meet your needs.

> You could still have permission issue, because by default this box creates shares under the user 'root'. 

I shared this, because the `/Users/dummy/project/tls` directory it will create, will per default (at least on my mac) create it with the `root` user, giving you permissions issue. Or you always chown it (crazy annoying) or you add in the beginning of your Vagrant file

``` ruby
config.nfs.map_uid = Process.uid   
config.nfs.map_gid = Process.gid 
```

From what I understand here is that this sets the user and group to the same one that runs the vagrant file.

Hope it helps :)

**Comment by thedug** [comment]
@tonivdv That does help. In your previous comment you an absolute path to /tls, which was confusing. :)

export DOCKER_CERT_PATH=/tls

**Comment by tonivdv** [comment]
@thedug Ah yes, sorry I removed my path and forget to set a dummy one :) My Bad. 

Glad it helped.

I'm probably going to do a PR for better doc on the  dduportal fork regarding all this.

**Comment by thedug** [comment]
Yes. I noticed a typo in that doc it is still referencing the upstream project.

Douglas

**Comment by Hellslicer** [comment]
In case someone still needs it : https://atlas.hashicorp.com/Hellslicer/boxes/boot2docker
It's the same box with Docker 1.5.0 instead of version 1.4.1.
